hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -1606,11 +1606,14 @@ def __init__(
         provider: Union[type, PrimitiveProvider] = HypothesisProvider,
         ir_prefix: Optional[Sequence[Union[NodeTemplate, ChoiceT]]] = None,
         max_length_ir: Optional[int] = None,
+        provider_kw: Optional[dict[str, Any]] = None,
     ) -> None:
         from hypothesis.internal.conjecture.engine import BUFFER_SIZE_IR
 
         if observer is None:
             observer = DataObserver()
+        if provider_kw is None:
+            provider_kw = {}
```

Comment:
Let's tack on an `elif not (isinstance(provider, type) and issubclass(provider, PrimitiveProvider): raise InvalidArgument(...)`